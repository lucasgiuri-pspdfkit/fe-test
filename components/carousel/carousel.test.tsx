import { render, screen, fireEvent } from "@testing-library/react";
import Carousel from "./carousel";

const mock2Items = [
  {
    id: "1",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 1",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 1",
    location: "Location 1",
  },
  {
    id: "2",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 2",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 2",
    location: "Location 2",
  },
];

const mockItems = [
  {
    id: "1",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 1",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 1",
    location: "Location 1",
  },
  {
    id: "2",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 2",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 2",
    location: "Location 2",
  },
  {
    id: "3",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 2",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 2",
    location: "Location 2",
  },
  {
    id: "4",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 2",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 2",
    location: "Location 2",
  },
  {
    id: "5",
    img: {
      src: "https://avatars.githubusercontent.com/u/1?v=4",
      alt: "Image 2",
      width: 100,
      height: 100,
    },
    date: {
      start: 123456789,
      end: 123456789,
    },
    title: "Title 2",
    location: "Location 2",
  },
];

describe("Carousel", () => {
  it("should render a carousel", () => {
    const view = render(<Carousel items={mockItems} />);
    expect(view).toMatchSnapshot();
  });

  it("should render a carousel with 2 items, arrow should not be visible", () => {
    render(<Carousel items={mock2Items} />);
    expect(screen.queryByRole("button")).not.toBeInTheDocument();
    expect(screen.getAllByRole("img").length).toBe(2);
  });

  it("should render a carousel with 2 items and click next", () => {
    render(<Carousel items={mockItems} />);
    fireEvent.click(screen.getByRole("button"));
    expect(screen.getAllByRole("img").length).toBe(3);
  });

  it("onClick should be called", () => {
    const onClick = jest.fn();
    render(<Carousel items={mockItems} onClick={onClick} />);
    fireEvent.click(screen.getByText("Title 1"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
