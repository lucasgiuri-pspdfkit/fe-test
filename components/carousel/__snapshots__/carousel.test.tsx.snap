// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render a carousel 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative h-full flex items-center"
      >
        <ul
          class="flex w-full"
        >
          <li
            class="w-full cursor-pointer pt-0 pb-0 shadow-md-bottom rounded-2xl"
          >
            <div
              class="flex items-center"
            >
              <img
                alt="Image 1"
                class="rounded-2xl overflow-hidden"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
              <div
                class="pl-[21px]"
              >
                <h4
                  class="text-lg
      font-avenirHeavy font-black
      text-gray-500"
                >
                  Title 1
                </h4>
                <p
                  class="whitespace-pre-line  text-gray-500"
                >
                  Location 1
                </p>
              </div>
            </div>
          </li>
          <li
            class="w-full cursor-pointer pt-0 pb-0 shadow-md-bottom rounded-2xl"
          >
            <div
              class="flex items-center"
            >
              <img
                alt="Image 2"
                class="rounded-2xl overflow-hidden"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75 2x"
                style="color: transparent;"
                width="100"
              />
              <div
                class="pl-[21px]"
              >
                <h4
                  class="text-lg
      font-avenirHeavy font-black
      text-gray-500"
                >
                  Title 2
                </h4>
                <p
                  class="whitespace-pre-line  text-gray-500"
                >
                  Location 2
                </p>
              </div>
            </div>
          </li>
        </ul>
        <button
          class=""
          type="button"
        >
          <img
            alt="Arrow right"
            class="rounded-none overflow-hidden"
            data-nimg="1"
            decoding="async"
            height="20"
            loading="lazy"
            src="/icons/arrow-right.svg"
            style="color: transparent;"
            width="20"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative h-full flex items-center"
    >
      <ul
        class="flex w-full"
      >
        <li
          class="w-full cursor-pointer pt-0 pb-0 shadow-md-bottom rounded-2xl"
        >
          <div
            class="flex items-center"
          >
            <img
              alt="Image 1"
              class="rounded-2xl overflow-hidden"
              data-nimg="1"
              decoding="async"
              height="100"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75 2x"
              style="color: transparent;"
              width="100"
            />
            <div
              class="pl-[21px]"
            >
              <h4
                class="text-lg
      font-avenirHeavy font-black
      text-gray-500"
              >
                Title 1
              </h4>
              <p
                class="whitespace-pre-line  text-gray-500"
              >
                Location 1
              </p>
            </div>
          </div>
        </li>
        <li
          class="w-full cursor-pointer pt-0 pb-0 shadow-md-bottom rounded-2xl"
        >
          <div
            class="flex items-center"
          >
            <img
              alt="Image 2"
              class="rounded-2xl overflow-hidden"
              data-nimg="1"
              decoding="async"
              height="100"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75"
              srcset="/_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Favatars.githubusercontent.com%2Fu%2F1%3Fv%3D4&w=256&q=75 2x"
              style="color: transparent;"
              width="100"
            />
            <div
              class="pl-[21px]"
            >
              <h4
                class="text-lg
      font-avenirHeavy font-black
      text-gray-500"
              >
                Title 2
              </h4>
              <p
                class="whitespace-pre-line  text-gray-500"
              >
                Location 2
              </p>
            </div>
          </div>
        </li>
      </ul>
      <button
        class=""
        type="button"
      >
        <img
          alt="Arrow right"
          class="rounded-none overflow-hidden"
          data-nimg="1"
          decoding="async"
          height="20"
          loading="lazy"
          src="/icons/arrow-right.svg"
          style="color: transparent;"
          width="20"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
